File: sdk/dotnet/lmsservice_db/src/api/learning.path/LearningPathModel.cs

using System.ComponentModel.DataAnnotations;

namespace lmsservice_db.src.api.learning.path.learning.pathmodel;
public class LearningPathCreateModel
{
        [Required]
    [StringLength(64, MinimumLength = 0)]
    public string TenantId { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string Name { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string Description { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string ImageUrl { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public int DurationInDays { get; set; } = 0;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public int PreferenceWeight { get; set; } = 0;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public bool Enabled { get; set; } = false;


}

public class LearningPathUpdateModel
{
        [Required]
    [StringLength(64, MinimumLength = 0)]
    public string TenantId { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string Name { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string Description { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string ImageUrl { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public int DurationInDays { get; set; } = 0;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public int PreferenceWeight { get; set; } = 0;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public bool Enabled { get; set; } = false;


}

public class LearningPathSearchFilters
{
        [StringLength(64, MinimumLength = 0)]
    public string TenantId { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string Name { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string Description { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string ImageUrl { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public int DurationInDays { get; set; } = 0;

    [StringLength(64, MinimumLength = 0)]
    public int PreferenceWeight { get; set; } = 0;

    [StringLength(64, MinimumLength = 0)]
    public bool Enabled { get; set; } = false;


}

public class ApiResponse
{
    [Required]
    public Dictionary<string, object>? Data { get; set; }

    [StringLength(500, MinimumLength = 2)]
    public string Message { get; set; } = string.Empty;

    public bool? Success { get; set; }
}

File: sdk/node/src/api/learning.path/learning.path.model.ts

import { IsString, IsNumber, IsOptional, IsBoolean, IsUUID } from 'class-validator';

export class LearningPathCreateModel {
    @IsOptional()
    @IsUUID()
    id?: string;  
    @IsString()
    TenantId: string;
    @IsString()
    Name: string;
    @IsString()
    Description: string;
    @IsString()
    ImageUrl: string;
    @IsNumber()
    DurationInDays: number;
    @IsNumber()
    PreferenceWeight: number;
    @IsString()
    Enabled: boolean;

}

export class LearningPathUpdateModel {
    @IsOptional()
    @IsString()
    TenantId?: string;
    @IsOptional()
    @IsString()
    Name?: string;
    @IsOptional()
    @IsString()
    Description?: string;
    @IsOptional()
    @IsString()
    ImageUrl?: string;
    @IsOptional()
    @IsNumber()
    DurationInDays?: number;
    @IsOptional()
    @IsNumber()
    PreferenceWeight?: number;
    @IsOptional()
    @IsString()
    Enabled?: boolean;

}

export class LearningPathSearchParams {
    @IsString()
    name: string;
}

export class ApiResponse<T> {
    Data: T;

    @IsOptional()
    @IsString()
    Message?: string;

    @IsOptional()
    @IsBoolean()
    Success?: boolean;
}